### PR TITLE
Add `api/v1/status/config` API

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
Add `api/v1/status/config` API support.
#### Which issue(s) this PR fixes or relates to

Ref https://github.com/grafana/mimir/issues/3323

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by hi-rustin** [comment]
Could you please take a look? Thanks! 💚 💙 💜 💛 ❤️

File: pkg/api/handlers.go [hunk 1]
```diff
@@ -161,6 +162,27 @@ func DefaultConfigHandler(actualCfg interface{}, defaultCfg interface{}) http.Ha
...
+type configResponse struct {
+	Status string `json:"status"`
+	Config string `json:"data"`
+}
+
+func (cfg *Config) statusConfigHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		data, err := yaml.Marshal(cfg)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (56quarters) [inline pkg/api/handlers.go:174]: Does the Prometheus API return errors in a particular response? e.g. `{"status":"error", "error":"YAML problems!"}`

File: pkg/api/handlers.go [hunk 2]
```diff
@@ -161,6 +162,27 @@ func DefaultConfigHandler(actualCfg interface{}, defaultCfg interface{}) http.Ha
...
 }
 
+type configResponse struct {
+	Status string `json:"status"`
+	Config string `json:"data"`
+}
+
+func (cfg *Config) statusConfigHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		data, err := yaml.Marshal(cfg)
```
Comment (56quarters) [inline pkg/api/handlers.go:172]: I don't think `cfg` is what you want to marshal here. `cfg` is the configuration for the `API` struct itself. This method will need to marshal `actualConfig` like the `/config` endpoint does.

Comment (hi-rustin) [reply pkg/api/handlers.go]: Fixed. Thanks!

File: pkg/api/handlers.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/api/handlers.go]: No, it just writes a string error to the response. Please see https://github.com/prometheus/prometheus/blob/c3fac587ef3c7d515e319844042ee0dddab54cbc/config/config.go#L252.

Comment (56quarters) [reply pkg/api/handlers.go]: Ah, it doesn't use `"error":err` but it looks like errors will still get wrapped in something like the `configResponse` struct you created here. Can you make sure we do the same thing?

https://github.com/prometheus/prometheus/blob/c3fac587ef3c7d515e319844042ee0dddab54cbc/web/api/v1/api.go#L1336-L1341

Comment (56quarters) [reply pkg/api/handlers.go]: Oh, nevermind. I see you've already done it!

**Review by 56quarters — APPROVED** [review]
LGTM, thanks!

**Pull request merged**